### ISSUE TITLE
Serve crashes on ParseError

### ISSUE THREAD

**Issue by rikhuijzer** [tlienart/Franklin.jl#715]
For most errors, Franklin will gracefully handle errors. For example, undefined variables are shown at the webpage. However, messing up the syntax, will crash the `serve` function completely.

An example of failing code is

    ```!
    f(x) = x
    f(2))
    ```

```
Base.Meta.ParseError("extra token \")\" after end of expression")
Stacktrace:
  ...
```

When in the middle of a webpage with some plots, this can be bothersome since restarting `serve` can take more than a minute. For me, this isn't the biggest priority. Still, it seemed a good idea to open an issue.

By the way,

    ```!
    f(x) = x
    f(2
    ```

fails gracefully and shows 
```
syntax: incomplete: premature end of input
```
in the webpage.

**Comment by tlienart** [comment]
> When in the middle of a webpage with some plots, this can be bothersome since restarting `serve` can take more than a minute.

I don't disagree with the rest of what you're saying but you're probably not using the right workflow here. Just start the REPL and leave it running!!! it should only take 1 second to get `serve()` going after the initial one. I typically start and stop mine quite often.

**Comment by rikhuijzer** [comment]
>  Just start the REPL and leave it running!!!

I'm using something like `julia -e 'using Pkg; ...'` out of convenience and because I thought it didn't matter. Apparently it does. Thank you again :smile: 😅

**Comment by tlienart** [comment]
lol, I'm impressed you stuck with it... that's really pretty slow if you only call that all the time 😱 

I'm reopening the issue because I do think something better could be done, like a red message on the page saying that the code failed. The only problem is that if cells further below require the output of that cell then... we could get weird things going on where it uses the previous eval results of that cell... anyway I'll think about it.

**Comment by rikhuijzer** [comment]
> lol, I'm impressed you stuck with it... that's really pretty slow if you only call that all the time :scream: (especially with plots, omg)

I have to admit that it did sometimes feel like a minefield where one typo would blow up the server. :laughing:

Sounds good :muscle:

**Comment by fredrikekre** [comment]
xref https://github.com/tlienart/Franklin.jl/issues/405. I think it would be reasonable if Frankling just `@warn`d, and then retried again on next save. For me, I seem to always hit save after typing the initial part of a code-block
````
```julia
````
which crashes with a parser error too.

**Comment by rikhuijzer** [comment]
As another case, Franklin can completely crash on some LaTeX. By *completely*, I mean that the REPL shuts down too which makes Franklin feel brittle (or like a minefield as I said above). Anyway, to reproduce, type
```
$$
\begin{aligned}
  h_i &\sim \text{Normal}(\mu, \sigma) \\
\end{aligned}
$$
```

**Comment by tlienart** [comment]
Yup thanks this is WIP, I'm doing a heavy refactoring of the parser which ultimately should help with

* user extending the parser
* system more tolerant to errors
* a path to PDF generation

I'm hoping this will be in by the end of January :)

**Comment by rikhuijzer** [comment]
Thanks for the update :+1:
